pandas/core/arrays/sparse/array.py

```diff
@@ -768,6 +770,9 @@ def __getitem__(self, key):
                 else:
                     key = np.asarray(key)
 
+            if is_list_like(key):
```

Comment:
is this repeated non purpose?

Reply:
repeated from where?

Reply:
the next check is_bool_indexer is duplicative

Reply:
It's not fully duplicative, see my long explanation at https://github.com/pandas-dev/pandas/pull/31150#issuecomment-577679134. It's mainly for dealing with object dtype.